posthog/api/feature_flag.py

```diff
@@ -54,8 +55,36 @@ class Meta:
             "experiment_set",
             "rollback_conditions",
             "performed_rollback",
+            "permission",
         ]
 
+    def get_permission(self, feature_flag: FeatureFlag) -> Organization.FeatureFlagsAccessLevel:
+        # TODO: make sure this isn't n+1
+        try:
+            from ee.models.feature_flag_role_access import FeatureFlagRoleAccess
+        except:
+            return Organization.FeatureFlagsAccessLevel.CAN_ALWAYS_EDIT
+        else:
+            request = self.context["request"]
+            all_role_memberships = request.user.role_memberships.select_related("role").all()
+            org_level = request.user.organization.feature_flags_access_level
+            role_level = max(
+                [membership.role.feature_flags_access_level for membership in all_role_memberships], default=0
+            )
+            final_level = max(role_level, org_level)
+            if final_level == Organization.FeatureFlagsAccessLevel.DEFAULT_VIEW_ALLOW_EDIT_BASED_ON_ROLE:
+                ff_access = FeatureFlagRoleAccess.objects.filter(
+                    feature_flag__id=feature_flag.pk,
+                    role__id__in=[membership.role.pk for membership in all_role_memberships],
+                ).exists()
+                return (
+                    Organization.FeatureFlagsAccessLevel.CAN_ALWAYS_EDIT
+                    if ff_access
+                    else Organization.FeatureFlagsAccessLevel.CAN_ONLY_VIEW
+                )
+            else:
+                return final_level
```

Comment:
why do we return a level here instead of just a boolean?

Reply:
Either works! Was planning with levels because there are multiple access levels but we could make it a field called `read_only: true/false`

Reply:
Let's do a `user_can_edit` or something like that and simplify it? Now the FE has one less layer of logic it needs to wrap around 😃

Reply:
sounds good—check the changes

Reply:
Either works, the only part I find not great about booleans in the frontend is that it becomes more complicated when we do end up adding more things to permissions:

1. You then need some way to understand that booleans mean these other access levels.
2. Need to take care of caching before you rip out the old stuff.

It's not too different for the frontend right now whether it's a bool or int, but a bool creates a few more problems down the line, since it's transforming a model that then needs to be transformed back 🤷 .

Reply:
(while if it stays as numbers right now, new numbers are just added seamlessly)

Reply:
It adds a bit more complexity on the FE which is fine but means you'll probably have to implement and maintain a duplicate system to determine access levels editing there, 

> since it's transforming a model that then needs to be transformed back

Why does turning it into a bool create more problems with ^? Isn't showing the "level" already adding transformation?

Reply:
D'oh, my bad, there's no caching since no results 😅 . Confused it with an insight result 🤦

Reply:
>Why does turning it into a bool create more problems with ^? Isn't showing the "level" already adding transformation?

I mean this in the sense that the backend model is based on levels, so if we send the level to the frontend, there's no transformation of the data model, but the bool is changing a level to a bool, which you need to revert when you add more levels because the transformation doesn't work anymore.